src/egamma_tnp/nanoaod_efficiency.py

```diff
@@ -512,6 +513,13 @@ def __repr__(self):
         return f"PhotonTagNProbeFromNanoAOD(Filters: {self.filters}, Number of files: {n_of_files})"
 
     def find_probes(self, events, cut_and_count, mass_range, vars):
+        # TODO: remove this temporary fix when https://github.com/scikit-hep/vector/issues/498 is resolved
+        photon_dict = {field: events.Photon[field] for field in events.Photon.fields} | {
+            "mass": dak.zeros_like(events.Photon.pt),
+            "charge": dak.zeros_like(events.Photon.pt),
+        }
+        events["Photon"] = dak.zip(photon_dict, with_name="Photon", behavior=candidate.behavior)
```

Comment:
@Saransh-cpp could you let me me know what behavior should I specify here. Maybe `coffea.nanoevents.methods import candidate` and `behavior=nanoaod.behavior`. This is me not understanding well how behaviors work. I don't even know if this does anything in this case, I'm just trying to make sure I'm not missing anything to make the new photons behave like they should.

Reply:
Hi! It should be `behavior=nanoaod.behavior`. `candidate.behavior` is included within `nanoaod.behavior`. coffea does not update the global awkward `behavior` (which is a good thing), which is why you need that extra `behavior=` arg.

Alternatively, to skip the `behavior` keyword, you could do something like -
```py
>>> ak.behavior.update(nanoaod.behavior)
>>> a = ak.zip({"x": [1], "y": [2], "z": [3], "m": [4]}, with_name="Photon")
>>> a
<PhotonArray [{x: 1, y: 2, z: 3, m: 4}] type='1 * Photon[x: int64, y: int64...'>
```

Makes me wonder if coffea should have something like [`vector.register_awkward`](https://github.com/scikit-hep/vector/blob/45bcb1a0fcc573d0326d5d1af0b6a2a4ad09a16b/src/vector/__init__.py#L177-L195) to make it easy for the users.